lib/logstash/outputs/scalyr.rb

```diff
@@ -991,6 +1004,46 @@ def add_client_timestamp_to_body(body)
     body[:client_timestamp] = current_time_millis.to_s
   end
 
+  # This should only be called on events that have already been determined to be too large
+  def truncate_event(event)
+    attrs = event[:attrs]
+    new_attrs = Hash.new
+    total_size = 0
+    priority_events = ["serverHost", "parser", "logfile", "severity", "message"]
```

Comment:
We could define this as a top / module level constants, and perhaps we want to call it something along the lines of "special / reserved" field names or similar.

Reply:
Depending on where this processing happens (either before or after the special event level serverHost field value is set), we may also need to include ``EVENT_LEVEL_SERVER_HOST_ATTRIBUTE_NAME`` constant / ``__origServerHost`` field name here.

https://github.com/scalyr/logstash-output-scalyr/blob/5a0f322a4dac533fa3fd350369362b500230280c/lib/logstash/outputs/scalyr.rb#L776

On a related note, do we also want to include ``thread`` + ``log`` here?

And IIRC, post processing (what gets sent to the sever side), severity field value is stored in ``sev`` field - https://github.com/scalyr/logstash-output-scalyr/blob/5a0f322a4dac533fa3fd350369362b500230280c/lib/logstash/outputs/scalyr.rb#L897

Reply:
> And IIRC, post processing (what gets sent to the sever side), severity field value is stored in sev field

That is correct. That field is also in the parent object, so if it is present I don't touch it.

> And also worth noting that some of those field values (top level serverHost, logs, etc.) are, in some / all cases, sent once as session level parameters and not as event level attributes.

also correct. I decided against touching anything in there since that isn't where the problem currently lies. For the sake of expediency, and to avoid introducing regression, I felt it better to be surgical in my approach.

Reply:
> Depending on where this processing happens (either before or after the special event level serverHost field value is set), we may also need to include EVENT_LEVEL_SERVER_HOST_ATTRIBUTE_NAME constant / __origServerHost field name here.

It happens after. I'll look into this. I think that data is stored outside of the `attrs` object, but I'll verify.

Reply:
Yeah, correct, some of those attributes (e.g. sev) are stored outside of the ``event.attrs`` (tbh, not sure how limits apply to those attributes).